mcodingbot/plugins/peps.py

```diff
@@ -20,11 +33,38 @@
 MAX_AGE_FOR_SEND = timedelta(minutes=1)
 MAX_AGE_FOR_EDIT = timedelta(minutes=5)
 
-recent_pep_responses: TTLCache[int, int] = TTLCache(
-    maxsize=100, ttl=MAX_AGE_FOR_EDIT.total_seconds()
+recent_pep_responses: TTLCache[int, MessageInfo] = TTLCache(
+    maxsize=10_000, ttl=MAX_AGE_FOR_EDIT.total_seconds()
 )
 plugin = Plugin()
 pep_manager = PEPManager()
+pep_cooldown: FixedMapping[PEPBucket] = FixedMapping(*CONFIG.pep_cooldown)
+
+
+def trigger_cooldowns(peps: Iterable[PEPInfo], channel_id: int) -> None:
+    for pep in peps:
+        pep_cooldown.trigger(PEPBucket(pep=pep.number, channel=channel_id))
+
+
+def reset_cooldowns(peps: Iterable[PEPInfo], channel_id: int) -> None:
+    for pep in peps:
+        pep_cooldown.reset(PEPBucket(pep=pep.number, channel=channel_id))
+
+
+def filter_can_send(
+    peps: Iterable[PEPInfo], channel_id: int
+) -> Iterable[PEPInfo]:
+    """
+    Removes peps that can not be sent because of the cooldown from the list
+    of peps.
+    """
+
+    def is_sendable(pep: PEPInfo) -> bool:
+        return pep_cooldown.can_trigger(
+            PEPBucket(pep=pep.number, channel=channel_id)
+        )
+
+    return filter(is_sendable, peps)
```

Comment:
Every instance of this function being called wraps it in a `set`.
Why not just return a set once instead.

```suggestion
def filter_can_send(
    peps: Iterable[PEPInfo], channel_id: int
) -> set[PEPInfo]:
    """
    Removes peps that can not be sent because of the cooldown from the list
    of peps.
    """

    def is_sendable(pep: PEPInfo) -> bool:
        return pep_cooldown.can_trigger(
            PEPBucket(pep=pep.number, channel=channel_id)
        )

    return set(filter(is_sendable, peps))
```

Reply:
The method is a filtering method. Not a convert to set method. Imo converting to a set may be too many things.

Reply:
that's not even a method 😶

Reply:
personally I think it makes sense for it to return an Iterable. Even though it is always converted to a set, that's not because the return value should necessarily be a set. Kinda like if you have a function that calculates something, and the only thing you do with it is print the result - you might still want to just return the value, to keep functionality seperate.

Reply:
exactly my thoughts